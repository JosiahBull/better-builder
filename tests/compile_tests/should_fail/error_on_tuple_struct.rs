// TODO: eventually this test case should pass.

use better_builder::BetterBuilder;

#[derive(BetterBuilder)]
struct Cart(
    String,
    u8,
    Option<u8>,
    Vec<String>,
);

fn main() {}
