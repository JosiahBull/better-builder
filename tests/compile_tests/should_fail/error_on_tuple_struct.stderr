error: BetterBuilder can only be derived on structs with named fields.
 --> tests/compile_tests/should_fail/error_on_tuple_struct.rs:4:1
  |
4 | / struct Cart(
5 | |     String,
6 | |     u8,
7 | |     Option<u8>,
8 | |     Vec<String>,
9 | | );
  | |__^
